### PR TITLE
Add token transfer to the client.

### LINKED ISSUE #320: Add multisig support for SPL token transfers
@FelixLutsch mentioned that for the incentive program, we may need the multisig to distribute wrapped LDO tokens. To support that, we need:

* A command to propose an SPL token transfer to the multisig
* A way to display it in `multisig show-transaction`

### PR DESCRIPTION
**Pull request by enriquefynn**
Add a new subcommand `token` to the multisig subcommand. In the future we may support more token instructions, e.g., burning, creating. For now we support only token transfers.

Note: when doing just the `multisig token` params, there's a clap error: "`Option::unwrap()` on a `None` value".
I'm still verifying what can be done, but it works well with the `--help` flag. 🤷 

Closes #320

### CODE REVIEW

File: cli/src/multisig.rs [hunk 1]
```diff
@@ -534,6 +574,24 @@ impl fmt::Display for ShowTransactionOutput {
...
+                match token_instruction {
+                    TokenInstruction::Transfer {
+                        from_address,
+                        to_address,
+                        amount,
+                    } => {
+                        writeln!(f, "It transfers tokens owned by the multisig.")?;
+                        writeln!(f, "    From address: {}", from_address)?;
+                        writeln!(f, "    To address:   {}", to_address)?;
+                        writeln!(f, "    Amount:       {}", amount)?;
```
Comment (ruuda) [inline cli/src/multisig.rs:588]: Would be good to clarify that this is the amount of the smallest denomination. E.g. if you do it for stSOL, it’s not the amount of stSOL but the amount of stLamports.

The mint also stores the number of decimals, maybe we can use that to format, but on the other hand, maybe that makes it ambiguous too :confused: 

Just clarifying that it is the smallest denomination should work. Also, should we print the token mint address of these accounts, for convenience?

File: program/tests/tests/withdrawals.rs [hunk 2]
```diff
@@ -4,6 +4,7 @@
 #![cfg(feature = "test-bpf")]
 
 use bincode::deserialize;
+use solana_program::program_error::ProgramError;
```
Comment (ruuda) [inline program/tests/tests/withdrawals.rs:7]: Is this change needed?

File: tests/util.py [hunk 3]
```diff
@@ -193,7 +193,7 @@ def create_vote_account(
     return test_account
 
 
-def create_spl_token(owner_keypair_fname: str, minter: str) -> str:
+def create_spl_token_account(owner_keypair_fname: str, minter: str) -> str:
```
Comment (ruuda) [inline tests/util.py:196]: :+1: the new name is much better!

**Review by ruuda — APPROVED** [review]
:+1: I have one small suggestion, but aside from that, this looks good to me.

Is the problem with the unwrap fixed? Nothing weird stood out to me in the PR.

File: cli/src/multisig.rs (hunk 1 above)
Comment (enriquefynn) [reply cli/src/multisig.rs:588]: I like printing the mint address, I'll add these changes.
Something like "Amount (in smallest denomination): " would work?

File: program/tests/tests/withdrawals.rs (hunk 2 above)
Comment (enriquefynn) [reply program/tests/tests/withdrawals.rs:7]: Good catch, I was testing some things and that was left.

File: cli/src/multisig.rs (hunk 1 above)
Comment (ruuda) [reply cli/src/multisig.rs:588]: :+1: yeah that works. If you like to keep then aligned, we could also do “Amount: X of the token’s smallest denomination”
